app/models/account/gateway.rb

```diff
@@ -88,4 +86,13 @@ def log_payment_gateway_change
                       "changed payment gateway to #{payment_gateway_type}")
   end
 
+  private
+
+  def build_payment_gateway
+    PaymentGateway.implementation(payment_gateway_type).new(payment_gateway_options)
+  rescue ArgumentError
```

Comment:
What is raising ArgumentError?

Reply:
This is tested in https://github.com/3scale/porta/pull/798/files#diff-c2ec270a6c4fc1a2ac8c06fa079eb49cR97
Remove this rescue and you will see that test failing 😄 

It happens when `payment_gateway_type` is `nil`.

https://github.com/3scale/porta/blob/4d587c90f31b8d38e1046dadc2f0c334c54ba98c/app/models/payment_gateway.rb#L60-L62

https://github.com/activemerchant/active_merchant/blob/24b99e8a57305d3ca826251b427930df9b3800a9/lib/active_merchant/billing/base.rb#L23-L40